src/main/java/com/questhelper/quests/thequeenofthieves/TheQueenOfThieves.java

```diff
@@ -0,0 +1,210 @@
+		upstairsHughesHouse = new Zone(new WorldPoint(1668,  3675, 1), new WorldPoint(1685, 3684, 1));
+	}
+
+	public void setupConditions()
+	{
+		inWarrens = new ZoneCondition(warrens);
+		inUpstairsHughesHouse = new ZoneCondition(upstairsHughesHouse);
+	}
+
+	public void setupSteps() {
+		WorldPoint tomasPoint = new WorldPoint(1796, 3781, 0);
+		WorldPoint manholePoint = new WorldPoint(1813, 3745, 0);
+		WorldPoint ladderPoint = new WorldPoint(1813, 10145, 0);
+		WorldPoint devanPoint = new WorldPoint(1767, 10146, 0);
+		WorldPoint queenOfThievesPoint = new WorldPoint(1764, 10158, 0);
+
+		talkToLawry = new NpcStep(this, NpcID.TOMAS_LAWRY, tomasPoint, "Speak to Tomas Lawry in Port Piscarilius.");
+		talkToLawry.addDialogStep("I'm looking for a quest.");
+		talkToLawry.addDialogStep("Absolutely.");
+
+		talkToPoorLookingPerson = new NpcStep(this, NpcID.POOR_LOOKING_WOMAN_7923, new WorldPoint(1803, 3738, 0), "Speak to the poor looking woman.");
+		talkToOReilly = new NpcStep(this, NpcID.ROBERT_OREILLY, new WorldPoint(1794, 3757, 0), "Speak to Robert O'Reilly, and give him the bowl of stew.");
+		talkToOReilly.addDialogStep("Okay.");
+
+		enterWarrens = new ObjectStep(this, ObjectID.MANHOLE_31706, manholePoint, "Enter the Warrens.");
+		enterWarrens.addAlternateObjects(ObjectID.MANHOLE_31707);
+
+		talkToDevan = new NpcStep(this, NpcID.DEVAN_RUTTER, devanPoint, "Speak to Devan Rutter.");
+
+		exitWarrens = new ObjectStep(this, ObjectID.LADDER_31708, ladderPoint, "Exit the Warrens.");
+
+		// Could potentially add a dialog step for Brutally or Softly, but seems unnecessary since either works.
+		killConrad = new NpcStep(this, NpcID.CONRAD_KING, new WorldPoint(1847, 3734, 0), "Murder Conrad King.");
+
+		// Enter the Warrens again.
+		enterWarrens2 = new ObjectStep(this, ObjectID.MANHOLE_31706, manholePoint, "Enter the Warrens.");
+		enterWarrens2.addAlternateObjects(ObjectID.MANHOLE_31707);
+
+		tellDevanAboutConrad = new NpcStep(this, NpcID.DEVAN_RUTTER, devanPoint, "Tell Devan Rutter about the murder.");
+
+		talkToQueenOfThieves = new NpcStep(this, NpcID.THE_QUEEN_OF_THIEVES, queenOfThievesPoint, "Talk to the Queen of Thieves.");
+		talkToQueenOfThieves.addAlternateNpcs(NpcID.LADY_SHAUNA_PISCARILIUS);
+
+		// Exit the Warrens again.
+		exitWarrens2 = new ObjectStep(this, ObjectID.LADDER_31708, ladderPoint, "Exit the Warrens.");
+
+		goToKingstown = new ObjectStep(this, ObjectID.STAIRCASE_11796, new WorldPoint(1672, 3681, 0), "Go up the stairs in Councillor Hughes' home in Kingstown.");
+		openChest = new ObjectStep(this, ObjectID.CHEST_31710, new WorldPoint(1681, 3677, 1), "Pick the locked chest.");
+
+		// This does not seem to properly highlight the object.
+		leaveKingtstown = new ObjectStep(this, ObjectID.STAIRCASE_11799, "Go downstairs.");
+
+		// Talk to Lawry again
+		talkToLawry2 = new NpcStep(this, NpcID.TOMAS_LAWRY, tomasPoint, "Speak to Tomas Lawry in Port Piscarilius.");
```

Comment:
This step requires the LETTER_21174. Would be good to have it as a requirement, and also add a tip to it stating something like 'You can get another from upstairs in Hughes' house in Kingstown'.

Reply:
Dialog highlight of 'Let's talk about my quests.' here would be good.

Reply:
ItemRequirement added in 1f59397 and tip added in d46b9c6.